device/iot-device-samples/send-event/src/main/java/samples/com/microsoft/azure/sdk/iot/InputParameters.java

```diff
@@ -0,0 +1,139 @@
+package samples.com.microsoft.azure.sdk.iot;
+
+import org.apache.commons.cli.*;
+
+import java.net.URISyntaxException;
+
+public class InputParameters
+{
+    private static final String PROTOCOL = "mqtt";
+    private static final String PATH_CERT = null;
+
+    private static final String FOOTER = "\nFor more info, please refer to https://github.com/Azure/azure-iot-sdks";
+    private static final String APPEXE = "java -jar ";
+
+    private static CommandLine cmd = null;
+
+    /**
+     * Setup parameters from command line arguments
+     * @param args array from main()
+     */
+    public InputParameters(String[] args){
+
+        // create option for cli input
+        Options options = new Options()
+                .addOption(
+                        Option.builder("h")
+                                .longOpt("help")
+                                .hasArg(false)
+                                .desc("Prints help message")
+                                .build()
+                )
+                .addOption(
+                        Option.builder("c")
+                                .longOpt("connectionString")
+                                .hasArg()
+                                .desc("IoT Hub or Edge Hub connection string (required; contains Hostname, Device Id & Device Key in one of the following formats: HostName=<iothub_host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key> or HostName=<iothub_host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key>;GatewayHostName=<gateway>)")
+                                .build()
+                )
+                .addOption(
+                        Option.builder("r")
+                                .longOpt("requests")
+                                .hasArg()
```

Comment:
`.hasArg()` does it have true default value?

Reply:
`.hasArg()` Indicates that the Option will require an argument. [links](https://commons.apache.org/proper/commons-cli/apidocs/org/apache/commons/cli/Option.Builder.html#hasArg--)

Reply:
> 

`.hasArg()` Indicates that the Option will require an argument. [links](https://commons.apache.org/proper/commons-cli/apidocs/org/apache/commons/cli/Option.Builder.html#hasArg--)

will this bring any confusion here?

Reply:
just trying to fix my ignorance :)